x/wasm-storage/keeper/keeper.go

```diff
@@ -35,139 +78,164 @@ func (k Keeper) GetAuthority() string {
+	for ; iter.Valid(); iter.Next() {
+		kv, err := iter.KeyValue()
+		if err != nil {
+			return err
+		}
 
-		if callback(wasm) {
+		if callback(kv.Value) {
 			break
 		}
 	}
+	return nil
 }
 
 // IterateAllOverlayWasms iterates over the all the stored Overlay Wasms
 // and performs a given callback function.
-func (k Keeper) IterateAllOverlayWasms(ctx sdk.Context, callback func(wasm types.Wasm) (stop bool)) {
-	store := ctx.KVStore(k.storeKey)
-	iterator := storetypes.KVStorePrefixIterator(store, types.KeyPrefixOverlay)
+func (k Keeper) IterateAllOverlayWasms(ctx sdk.Context, callback func(wasm types.Wasm) (stop bool)) error {
+	iter, err := k.OverlayWasm.Iterate(ctx, nil)
+	if err != nil {
+		return err
+	}
+	defer iter.Close()
 
-	defer iterator.Close()
-	for ; iterator.Valid(); iterator.Next() {
-		var wasm types.Wasm
-		k.cdc.MustUnmarshal(iterator.Value(), &wasm)
+	for ; iter.Valid(); iter.Next() {
+		kv, err := iter.KeyValue()
+		if err != nil {
+			return err
+		}
 
-		if callback(wasm) {
+		if callback(kv.Value) {
 			break
 		}
 	}
+	return nil
 }
 
 // ListDataRequestWasms returns hashes and types of all Data Request Wasms
 // in the store.
 func (k Keeper) ListDataRequestWasms(ctx sdk.Context) []string {
 	var hashTypePairs []string
-	k.IterateAllDataRequestWasms(ctx, func(w types.Wasm) bool {
+	err := k.IterateAllDataRequestWasms(ctx, func(w types.Wasm) bool {
 		hashTypePairs = append(hashTypePairs, hex.EncodeToString(w.Hash)+","+w.WasmType.String())
 		return false
 	})
+	if err != nil {
+		return nil
+	}
 	return hashTypePairs
 }
 
 // ListOverlayWasms returns hashes and types of all Overlay Wasms in the store.
 func (k Keeper) ListOverlayWasms(ctx sdk.Context) []string {
 	var hashTypePairs []string
-	k.IterateAllOverlayWasms(ctx, func(w types.Wasm) bool {
+	err := k.IterateAllOverlayWasms(ctx, func(w types.Wasm) bool {
 		hashTypePairs = append(hashTypePairs, hex.EncodeToString(w.Hash)+","+w.WasmType.String())
 		return false
 	})
+	if err != nil {
+		return nil
+	}
 	return hashTypePairs
 }
 
 func (k Keeper) GetAllWasms(ctx sdk.Context) []types.Wasm {
 	var wasms []types.Wasm
-	k.IterateAllDataRequestWasms(ctx, func(wasm types.Wasm) bool {
+	err := k.IterateAllDataRequestWasms(ctx, func(wasm types.Wasm) bool {
 		wasms = append(wasms, wasm)
 		return false
 	})
-	k.IterateAllOverlayWasms(ctx, func(wasm types.Wasm) bool {
+	if err != nil {
+		return nil
+	}
+	err = k.IterateAllOverlayWasms(ctx, func(wasm types.Wasm) bool {
 		wasms = append(wasms, wasm)
 		return false
 	})
+	if err != nil {
+		return nil
+	}
 	return wasms
 }
 
+// SetProxyContractRegistry stores Proxy Contract address.
+func (k Keeper) SetProxyContractRegistry(ctx sdk.Context, address sdk.AccAddress) error {
+	return k.ProxyContractRegistry.Set(ctx, address)
+}
+
+// GetProxyContractRegistry returns Proxy Contract address.
+func (k Keeper) GetProxyContractRegistry(ctx sdk.Context) (sdk.AccAddress, error) {
+	return k.ProxyContractRegistry.Get(ctx)
+}
+
```

Comment:
We don't need these functions anymore.

Reply:
Refactored here [3310202](https://github.com/sedaprotocol/seda-chain/pull/192/commits/3310202f76cc5b82224800e2233e934a90f589f1)